apps/yapms/src/lib/components/modals/editregionmodal/EditRegionModal.svelte

```diff
@@ -8,9 +8,8 @@
 	$: value = open ? $EditRegionModalStore.region?.permaVal : undefined;
 	$: newValue = value ?? 0;
 
-	$: if (newValue < 0 || newValue === null) {
-		newValue = 0;
-	}
```

Comment:
Are you against preventing the user from entering invalid data here?

Reply:
I am against modifying the contents of newValue directly. Under that behavior, whenever the user deletes the current value to enter in a new value, a 0 will be put in that they can't remove.
![yapmsZeroInsert](https://github.com/yapms/yapms/assets/42476312/ed2553e8-4a58-4cd1-a70a-ed2007835f06)

Preventing invalid data from being submitted with an explanation of why that data is invalid seems like the better solution to me.

Reply:
Oh didn't notice that.

What if we just allowed nulls, and treated it as 0 when the user submitted?

Reply:
That sounds good. Will change

Reply:
I was thinking to keep the behavior of preventing the user from entering negative numbers. Are you against that?

Reply:
Also, I am able to type random stuff into the box and it allows it.

![image](https://github.com/yapms/yapms/assets/13600696/7cb4070d-ade4-49a8-827e-8ff83c63bba0)

Reply:
I will fix random stuff, imo having a descriptive error message telling the user what is wrong when they enter in a negative number is good. We could change it to warning text that says "negatives are not supported" and then when it is submitted just take the absolute value.

Reply:
Alright then, the current message you have is fine.